core/domain/state_domain.py

```diff
@@ -737,13 +737,14 @@ def validate(self):
             raise utils.ValidationError(
                 'Expected needs_update to be a bool, received %s' %
                 self.needs_update)
-        if not isinstance(self.duration_secs, float):
+        if not isinstance(self.duration_secs, (float, int)):
             raise utils.ValidationError(
-                'Expected duration_secs to be a float, received %s' %
+                'Expected duration_secs to be a float or int, received %s' %
                 self.duration_secs)
-        if self.duration_secs <= 0:
+        if self.duration_secs < 0:
             raise utils.ValidationError(
-                'Invalid duration in seconds: %s' % self.duration_secs)
+                'Expected duration_secs to be positive number %s' %
```

Comment:
Message should also say "or zero, if not yet specified"?

Reply:
ok changed the message to "Expected duration_secs to be positive number or zero, if not yet specified"

Reply:
Done.